### ISSUE TITLE
When will upload via chunks be available?

### ISSUE THREAD

**Issue by rugdealer** [websemantics/vimeo-upload#11]
I see many TODOs in your code for future integration of an upload via chunks. When do you plan on integrating it? Thanks.

**Comment by websemantics** [comment]
I believe chunks upload is to enable pause / resume and not sure if that would solve uploading large files. Don't have plans for implementing this atm, put would welcome any contributions,

**Comment by rugdealer** [comment]
Hey, thanks for the response. I'll look into it and if I find a solution I'll contribute.
Also wanted to know if there is a way to have it upload in chunks by default to be able to test to see if the error still shows up?

**Comment by rugdealer** [comment]
@websemantics I forgot to get back to you, but I created a typescript project based off of your vimeo-upload. It uploads purely by chunks (and updates the chunk size based on the connection speed). If you're interested in using it, I could eventually submit a pull request. It could be seen a v2 of your original project. However, I'd have to change a few things to make it more dynamic.

**Comment by websemantics** [comment]
Hi @rugdealer,
That would be a great contribution initially as a usage example then as an integral part of the plugin. Does this have a progress indicator? as a callback function maybe? 
All the best,

**Comment by rugdealer** [comment]
@websemantics 
Yes, it actually has two progress indicators. One for the chunk upload and one for the total upload. You can check out a version I pushed two months ago https://github.com/rugdealer/vimeo-upload . I changed a few things for the company I work for.

I'll prepare an easier version to use. I'll keep you posted.

**Comment by jehartzog** [comment]
I am also using a slightly modified version of vimeo-upload in production and am very interested in the ability to upload via chunks. Right now not all my client's large file uploads go through, most often on the clients with poorer internet connections.

I'd be happy to put together a PR, let me know.

**Comment by websemantics** [comment]
I will look forward to your PR @rugdealer, this is exciting 👍 
@jehartzog hi, do you want to wait until we have @rugdealer PR? it sounds fully featured,

**Comment by jehartzog** [comment]
I'll definitely wait, but just letting you know I'll be testing and possibly pushing to production when ready, so can help out whenever needed.

**Comment by rugdealer** [comment]
I've modified quite a bit with the project. I'm currently debugging and I expect the first version of the project to be finished in a few days. How should we handle the PR? Should a branch be created? The tricky part is, it's a project started from scratch so it uses a completely different .git. Should the project be completely separate and renamed "vimeo-chunk-upload" ? Please let me know what you think would be best @websemantics thanks.

**Comment by websemantics** [comment]
@rugdealer, I don't mind having own fork `vimeo-chunk-upload` and I'll surely link it from README.md for the benefit of interested developers,

I've created a new branch`chunk-upload` in case you'd like to integrate with the current codebase (fork, update, PR). Like you mentioned, adding docs and updating demo code for other developers to fully take advantage of this new feature is preferred,

If you do go with the later, make sure you add yourself to the authors list (i.e. package.json etc) as this is a major contribution and since your help will also be needed to fix any bugs :)

**Comment by rugdealer** [comment]
@websemantics ok great. I've forked the project and started adding the changes. I'll send the pull request in a few days to your `chunk-upload` branch.
